go/engine/device_keygen.go

```diff
@@ -412,13 +401,58 @@ func (e *DeviceKeygen) prepareSDHBoxesFromPaperkey(ctx *Context) ([]keybase1.Sha
 	if err != nil {
 		return nil, err
 	}
-	err = sdhk.SyncAsPaperKey(ctx.NetContext, ctx.LoginContext, &upak, paperDeviceID, paperEncKey)
+	err = pukring.SyncAsPaperKey(ctx.NetContext, ctx.LoginContext, &upak, paperDeviceID, paperEncKey)
 	if err != nil {
 		return nil, err
 	}
-	sdhBoxes, err := sdhk.PrepareBoxesForNewDevice(ctx.NetContext,
+	pukBox, err := pukring.PrepareBoxForNewDevice(ctx.NetContext,
 		e.EncryptionKey(), // receiver key: provisionee enc
 		paperEncKey,       // sender key: paper key enc
 	)
-	return sdhBoxes, err
+	return []keybase1.PerUserKeyBox{pukBox}, err
+}
+
+func (e *DeviceKeygen) makePerUserKeySig(me *libkb.User, pukSeed libkb.PerUserKeySeed, signer libkb.GenericKey) (libkb.JSONPayload, error) {
+	gen := keybase1.PerUserKeyGeneration(1)
+
+	pukSigKey, err := pukSeed.DeriveSigningKey()
+	if err != nil {
+		return nil, err
+	}
+
+	pukEncKey, err := pukSeed.DeriveEncryptionKey()
+	if err != nil {
+		return nil, err
+	}
+
+	// Make reverse sig
+	jwRev, err := libkb.PerUserKeyProof(me, pukSigKey.GetKID(), pukEncKey.GetKID(), gen, signer, nil)
+	if err != nil {
+		return nil, err
+	}
+	reverseSig, _, _, err := libkb.SignJSON(jwRev, pukSigKey)
+	if err != nil {
+		return nil, err
+	}
+
+	// Make sig
+	jw, err := libkb.PerUserKeyProof(me, pukSigKey.GetKID(), pukEncKey.GetKID(), gen, signer, &reverseSig)
+	if err != nil {
+		return nil, err
+	}
+	sig, _, _, err := libkb.SignJSON(jw, signer)
+	if err != nil {
+		return nil, err
+	}
+
+	publicKeysEntry := make(libkb.JSONPayload)
+	publicKeysEntry["signing"] = pukSigKey.GetKID().String()
+	publicKeysEntry["encryption"] = pukEncKey.GetKID().String()
+
+	res := make(libkb.JSONPayload)
+	res["sig"] = sig
+	res["signing_kid"] = signer.GetKID().String()
+	res["type"] = libkb.LinkTypePerUserKey
+	res["public_keys"] = publicKeysEntry
```

Comment:
@maxtaco just checking, is this a legacy PGP-era sort of thing, where we're keeping up with it to make it easier to share server code?